tests/transport/spdp/spdp_transport.cpp

```diff
@@ -330,47 +348,42 @@ bool run_test()
   OpenDDS::DCPS::LocatorSeq nonEmptyList(1);
   nonEmptyList.length(1);
   nonEmptyList[0].port = 12345;
-
   nonEmptyList[0].kind = LOCATOR_KIND_UDPv4;
   std::memset(nonEmptyList[0].address, 0, 12);
   nonEmptyList[0].address[12] = 127;
   nonEmptyList[0].address[13] = 0;
   nonEmptyList[0].address[14] = 0;
   nonEmptyList[0].address[15] = 1;
 
+  OpenDDS::DCPS::LocatorSeq unicastLocators(2);
+  set_unicast_locators(unicastLocators, nonEmptyList[0]);
+
   const OpenDDS::RTPS::SPDPdiscoveredParticipantData pdata = {
-    {
-      DDS::BuiltinTopicKey_t(),
-      qos.user_data
-    },
+    {DDS::BuiltinTopicKey_t(), qos.user_data},
     {
       domain
       , ""
       , PROTOCOLVERSION
-      , {gp[0], gp[1], gp[2], gp[3], gp[4], gp[5],
-       gp[6], gp[7], gp[8], gp[9], gp[10], gp[11]}
+      , {gp[0], gp[1], gp[2], gp[3], gp[4], gp[5], gp[6], gp[7], gp[8], gp[9], gp[10], gp[11]}
       , VENDORID_OPENDDS
-      , false /*expectsIQoS*/
+      , false // expectsIQoS
       , availableBuiltinEndpoints
       , 0
-      , nonEmptyList /* sedp_multicast */
-      , nonEmptyList /* sedp_unicast */
-      , nonEmptyList /*defaultMulticastLocatorList*/
-      , nonEmptyList /*defaultUnicastLocatorList*/
-      , { 0 /*manualLivelinessCount*/ }
+      , unicastLocators // metatrafficUnicastLocatorList
+      , nonEmptyList    // metatrafficMulticastLocatorList
+      , nonEmptyList    // defaultMulticastLocatorList
+      , nonEmptyList    // defaultUnicastLocatorList
+      , {0} // manualLivelinessCount
       , qos.property
       , {PFLAGS_THIS_VERSION} // opendds_participant_flags
       , false // opendds_rtps_relay_application_participant
 #ifdef OPENDDS_SECURITY
       , availableExtendedBuiltinEndpoints
 #endif
     },
-    { // Duration_t (leaseDuration)
-      static_cast<CORBA::Long>((rd.resend_period() * 10).value().sec()),
-      0 // we are not supporting fractional seconds in the lease duration
-    },
-    { 0, 0},
-    0
+    // Duration_t (leaseDuration - we are not supporting fractional seconds)
+    {static_cast<CORBA::Long>((rd.resend_period() * 10).value().sec()), 0},
```

Comment:
I don't think this change improves anything. The comment about fractional seconds is now in a less helpful position.